index.bs

```diff
@@ -2619,67 +2619,66 @@ structures.
 
 ## Registering a new credential ## {#registering-a-new-credential}
 
-When registering a new credential, represented by a {{AuthenticatorAttestationResponse}} structure, as part of a
+When registering a new credential, represented by a {{AuthenticatorAttestationResponse}} structure |response|, as part of a
 [=registration=] [=ceremony=], a [=[RP]=] MUST proceed as follows:
 
-1. Perform JSON deserialization on the {{AuthenticatorResponse/clientDataJSON}} field of the
-    {{AuthenticatorAttestationResponse}} object to extract the [=client data=] |C| claimed as collected during the credential
-    creation.
+1. Let |C|, the [=client data=] claimed as collected during the credential creation, be the result of [=parse JSON from
+    bytes|parsing JSON from the bytes=] in <code>|response|.{{AuthenticatorResponse/clientDataJSON}}</code>.
 
-2. Verify that the {{CollectedClientData/type}} in |C| is the string `webauthn.create`.
+1. Verify that the {{CollectedClientData/type}} in |C| is the string `webauthn.create`.
 
-2. Verify that the {{CollectedClientData/challenge}} in |C| matches the challenge that was sent to the authenticator in the
+1. Verify that the {{CollectedClientData/challenge}} in |C| matches the challenge that was sent to the authenticator in the
     {{CredentialsContainer/create()}} call.
 
-3. Verify that the {{CollectedClientData/origin}} in |C| matches the [=[RP]=]'s [=origin=].
+1. Verify that the {{CollectedClientData/origin}} in |C| matches the [=[RP]=]'s [=origin=].
 
-4. Verify that the {{CollectedClientData/tokenBindingId}} in |C| matches the [=Token Binding ID=] for the TLS connection over
+1. Verify that the {{CollectedClientData/tokenBindingId}} in |C| matches the [=Token Binding ID=] for the TLS connection over
     which the attestation was obtained.
 
-5. Verify that the {{CollectedClientData/clientExtensions}} in |C| is a subset of the extensions requested by the RP
+1. Verify that the {{CollectedClientData/clientExtensions}} in |C| is a subset of the extensions requested by the RP
     and that the {{CollectedClientData/authenticatorExtensions}} in |C| is also a subset of the extensions requested by
     the RP.
 
-6. Compute the hash of {{AuthenticatorResponse/clientDataJSON}} using SHA-256.
+1. Compute the hash of <code>|response|.{{AuthenticatorResponse/clientDataJSON}}</code> using SHA-256.
 
-7. Perform CBOR decoding on the {{AuthenticatorAttestationResponse/attestationObject}} field of the
+1. Perform CBOR decoding on the {{AuthenticatorAttestationResponse/attestationObject}} field of the
     {{AuthenticatorAttestationResponse}} structure to obtain the attestation statement format |fmt|, the [=authenticator data=]
     |authData|, and the attestation statement |attStmt|.
 
-8. Verify that the [=RP ID=] hash in |authData| is indeed the SHA-256 hash of the [=RP ID=] expected by the RP.
+1. Verify that the [=RP ID=] hash in |authData| is indeed the SHA-256 hash of the [=RP ID=] expected by the RP.
```

Comment:
suggest:
> Verify that \<code>|authData|.rpIdHash\</code> is indeed the SHA-256 hash of the [=RP ID=] expected by the RP.

..?

Reply:
My thinking in writing this one on the "the X in |authData|" format was to hint at the fact that the representation of |authData| doesn't separate the semantic contents into neat members like JSON or CBOR does. What do you think of that? I'm not against changing it if you still think it's better to write it on the "|authData|.X" format.